src/Dependencies/Collections/Internal/SegmentedArrayHelper.cs

```diff
@@ -19,52 +19,64 @@ internal static class SegmentedArrayHelper
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static int GetSegmentSize<T>()
         {
+            return Unsafe.SizeOf<T>() switch
+            {
+                4 => 16384,
```

Comment:
Perhaps worth a comment explaining that these are just the most popular values?

Reply:
➡️ Comments added